src/check.h.in

```diff
@@ -669,6 +671,263 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
+ * compared to ck_assert(). */
+/* OP may be any comparison operator. */
+#define _ck_assert_floating(X, OP, Y) do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  ck_assert_msg(_ck_x OP _ck_y, "Assertion '%s' failed: %s == %.*Lg, %s == %.*Lg", \
+  #X" "#OP" "#Y, \
+  #X, (int)CK_FLOATING_DIG, _ck_x, \
+  #Y, (int)CK_FLOATING_DIG, _ck_y); \
+} while (0)
+
+/**
+ * Check two floating point numbers to determine if X==Y
+ *
+ * If not X==Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq(X, Y) _ck_assert_floating(X, ==, Y)
+/**
+ * Check two floating point numbers to determine if X!=Y
+ *
+ * If not X!=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ne(X, Y) _ck_assert_floating(X, !=, Y)
+/**
+ * Check two floating point numbers to determine if X<Y
+ *
+ * If not X<Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_lt(X, Y) _ck_assert_floating(X, <, Y)
+/**
+ * Check two floating point numbers to determine if X<=Y
+ *
+ * If not X<=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_le(X, Y) _ck_assert_floating(X, <=, Y)
+/**
+ * Check two floating point numbers to determine if X>Y
+ *
+ * If not X>Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_gt(X, Y) _ck_assert_floating(X, >, Y)
+/**
+ * Check two floating point numbers to determine if X>=Y
+ *
+ * If not X>=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ge(X, Y) _ck_assert_floating(X, >=, Y)
+
+/**
+ * Check precision P of floating point numbers for correctness.
+ *
+ * If not P >= 0 or not P <= |LDBL_MIN_10_EXP|, the test fails.
+ *
+ * @param P positive number of decimal places to be compared (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_prec(PREC) \
+do { \
+  int _ck_p = (PREC); \
+  int _prec = -_ck_p; \
+  ck_assert_msg(_prec <= 0, \
```

Comment:
This is a little confusing because the documentation states P is positive but the checks themselves use -P. I do not see the value in negating PREC in these checks. Could the initial value of PREC be used. This may help the user if an assertion does fail.